### DISCUSSION TITLE
Could Nextflow be adapted for Aliyun? Or otherwise, how can I develop Executors on cloud my self?

### DISCUSSION THREAD

**Discussion by wangshun1121** [nextflow-io/nextflow#2795] [Q&A] [answered]
I am a bioinformatics engineer in China, and I tried to deploy my nextflow pipeline to local cloud such as [Aliyun](https://www.alibabacloud.com/en?spm=5176.19720258.J_3207526240.exp-location-0.e9392c4aXn17p2), [Huawei Cloud](https://www.huaweicloud.com/intl/en-us/), [Tencent Cloud](https://intl.cloud.tencent.com/?lang=en&pg=)...  I wander if Nextflow can be adapted for these clouds like AWS etc...

Or otherwise, is there any templates for Nextflow Executors development? For example, google cloud, so that I can try to develop Executors for local cloud myself.

**Comment by pditommaso** [comment, marked as answer]
That's very welcome. If I remember correctly Alibaba Cloud Batch service has a very similar API to the Azure one. You may want to have a look at the [nf-azure](https://github.com/nextflow-io/nextflow/blob/3c4032d861368d84b98f7c28876d5b2f3b1c5c69/plugins/nf-azure/src/main/nextflow/cloud/azure#L312-L312) plugin to get started with it

**Reply by wangshun1121** [reply]
I'll try

**Comment by bentsherman** [comment]
Hi @wangshun1121 , another option is to setup a Kubernetes cluster on one of these clouds and use Nextflow's `k8s` executor. That way you don't have to develop a new nextflow executor, but you also don't get the benefit of auto-scaling that most cloud batch executors provide, unless one of these cloud providers have a managed Kubernetes service that support node auto-scaling (like Google GKE for example).

**Comment by denny-zhao** [comment]
Hi @wangshun1121 , we are contributing a new nextflow plugin which provides cloud integration automation. Alicloud is fully supported. You can reference to this disuccsion: nextflow-io/plugins#12
